### PR TITLE
Implement new sound system

### PR DESCRIPTION
**Pull request by J3fftw1**
## Description

Merged all commits from master
Updated some minor things that java 16 added
Please squash this pr because of all those commits
Changed more sounds over to the new system

## Proposed changes

Moved almost all sounds to the new system
Merged all commits from master
Updated some minor things that java 16 added

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [ ] I have made sure that the proposed changes do not break compatibility across the supported Minecraft versions (1.14.* - 1.19.*).
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

## Note
Build will fail since mockbuckit doesnt support strings as sound as cookie stated in #2844.
This however does build completely fine

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/armor/LongFallBoots.java [hunk 1]
```diff
@@ -10,6 +10,7 @@
 import io.github.thebusybiscuit.slimefun4.api.items.ItemGroup;
 import io.github.thebusybiscuit.slimefun4.api.items.SlimefunItemStack;
 import io.github.thebusybiscuit.slimefun4.api.recipes.RecipeType;
+import io.github.thebusybiscuit.slimefun4.core.services.sounds.SoundEffect;
```
Comment (JustAHuman-xD) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/armor/LongFallBoots.java:13]: duplicate import
```suggestion
```

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/entities/IronGolemAssembler.java [hunk 2]
```diff
@@ -13,6 +13,7 @@
 import io.github.thebusybiscuit.slimefun4.api.items.ItemGroup;
 import io.github.thebusybiscuit.slimefun4.api.items.SlimefunItemStack;
 import io.github.thebusybiscuit.slimefun4.api.recipes.RecipeType;
+import io.github.thebusybiscuit.slimefun4.core.services.sounds.SoundEffect;
```
Comment (JustAHuman-xD) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/entities/IronGolemAssembler.java:16]: duplicate import
```suggestion
```

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/magical/InfusedHopper.java [hunk 3]
```diff
@@ -19,6 +19,7 @@
 import io.github.thebusybiscuit.slimefun4.api.items.settings.DoubleRangeSetting;
 import io.github.thebusybiscuit.slimefun4.core.services.sounds.SoundEffect;
 import io.github.thebusybiscuit.slimefun4.api.recipes.RecipeType;
+import io.github.thebusybiscuit.slimefun4.core.services.sounds.SoundEffect;
```
Comment (JustAHuman-xD) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/magical/InfusedHopper.java:22]: duplicate import
```suggestion
```

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/Compressor.java [hunk 4]
```diff
@@ -93,11 +94,12 @@ private void craft(Player p, ItemStack output, Block dispenser, Inventory dispIn
...
-                } else {
-                    p.getWorld().playSound(p.getLocation(), Sound.ENTITY_ARROW_HIT_PLAYER, 1F, 1F);
-                    handleCraftedItem(output, dispenser, dispInv);
-                }
+                    if (j == 1) {
+                        SoundEffect.COMPRESSOR_CRAFT_CONTRACT_SOUND.playFor(p);
+                    } else {
+                        SoundEffect.COMPRESSOR_CRAFT_EXTEND_SOUND.playFor(p);
+                    }
+                 }
```
Comment (JustAHuman-xD) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/Compressor.java:102]: Won't this break the compressor??

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/miner/MiningTask.java [hunk 5]
```diff
@@ -199,7 +200,8 @@ public void run() {
 
                     if (miner.canMine(b) && push(miner.getOutcome(b.getType()))) {
                         furnace.getWorld().playEffect(furnace.getLocation(), Effect.STEP_SOUND, b.getType());
```
Comment (JustAHuman-xD) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/miner/MiningTask.java:202]: should we add the comment that we have all around for these step sounds

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/weapons/VampireBlade.java [hunk 6]
```diff
@@ -17,6 +17,7 @@
 import io.github.thebusybiscuit.slimefun4.core.services.sounds.SoundEffect;
 import io.github.thebusybiscuit.slimefun4.api.recipes.RecipeType;
 import io.github.thebusybiscuit.slimefun4.core.handlers.WeaponUseHandler;
+import io.github.thebusybiscuit.slimefun4.core.services.sounds.SoundEffect;
```
Comment (JustAHuman-xD) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/weapons/VampireBlade.java:20]: duplicate import
```suggestion
```

File: src/test/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TestFireworksListener.java [hunk 7]
```diff
@@ -21,7 +21,7 @@
 class TestFireworksListener {
 
     private static ServerMock server;
-
+    private static Slimefun plugin;
```
Comment (JustAHuman-xD) [inline src/test/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TestFireworksListener.java:24]: ?

File: src/test/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TestSmithingTableListener.java [hunk 8]
```diff
@@ -31,7 +31,6 @@ class TestSmithingTableListener {
 
     private static SmithingTableListener listener;
     private static ServerMock server;
-
```
Comment (JustAHuman-xD) [inline src/test/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TestSmithingTableListener.java:34]: ?

**Review by JustAHuman-xD — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/miner/MiningTask.java [hunk 9]
```diff
@@ -197,9 +198,11 @@ public void run() {
                         return;
                     }
 
-                    if (miner.canMine(b) && push(miner.getOutcome(b.getType()))) {
+                    if (miner.canMine(b) && push(miner.getOutcome(b.getType()))) {\
```
Comment (JustAHuman-xD) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/miner/MiningTask.java]:
```suggestion
                    if (miner.canMine(b) && push(miner.getOutcome(b.getType()))) {
```

**Comment by JustAHuman-xD** [comment]
again, i tried to cancel the review and it submitted it

**Comment by JustAHuman-xD** [comment]
ignore that

**Review by JustAHuman-xD — APPROVED** [review]
Github is annoying

**Review by WalshyDev — APPROVED** [review]
Let's hope things don't burn down :)

**Comment by J3fftw1** [comment]
> Let's hope things don't burn down :)

x

**Pull request merged**